app/code/Magento/Backend/Api/Search/ItemsInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Api\Search;
+
+/**
+ * @api
+ */
+interface ItemsInterface
+{
+    const LIMIT = 'limit';
+    const START = 'start';
+    const QUERY = 'query';
+
+    /**
+     * get the search result items
+     *
+     * @return array
+     */
+    public function getResults();
+
+    /**
+     * set offset
+     *
+     * @param int $start
+     * @return ItemsInterface
```

Comment:
@tzyganu  I think the existing SearchCriteria we have in Framework \Magento\Framework\Api\SearchCriteria would not fit your needs. It's too heavyweight for your specific case. 
When I said "SearchCriteria" I meant similar approach we use with SearchCriteria:

1. To have a dedicated interface to describe your request (aka SearchCriteria). In your case that would be something like:
```php
namespace Your\Custom\Api;

class SearchCriteria
{
      public function setStart($start);
      public function setQuery($query);
      public function setLimit($limit);
      public function getStart();
      public function getQuery();
      public function getLimit();
}
```

2. To have Search interfaces which will process a search query based on the SearchCriteria given as Input parameter. Something like:
```php
namespace Your\Custom\Api;

class QueryProcessor
{
      public function getResults(SearchCriteria $searchCriteria);  // or maybe Search(SearchCriteria $searchCriteria)
}
```

Doing so we make a responsibility segregation and don't have problems with managing and keeping the state in entity responsible for search request execution because all the parameters come as input from the outside each time.